itest/rust/src/builtin_tests/geometry/quaternion_test.rs

```diff
@@ -178,4 +178,28 @@ fn quaternion_spherical_cubic_interpolate_in_time() {
     );
     assert_eq!(outcome, Quaternion::default())
 }
+
+#[itest]
+fn quaternion_mul1() {
+    use godot::builtin::real;
+    use std::f32::consts::PI;
+
+    let q = Quaternion::from_axis_angle(Vector3::UP, PI / 2.0 as real);
+    let rotated = q * Vector3::new(1.0, 4.2, 0.0);
+    assert_eq_approx!(rotated.x, 0.0);
+    assert_eq_approx!(rotated.y, 4.2);
+    assert_eq_approx!(rotated.z, -1.0);
+}
+
+#[itest]
+fn quaternion_mul2() {
+    use godot::builtin::real;
+    use std::f32::consts::PI;
+
+    let q = Quaternion::from_axis_angle(-Vector3::UP, PI / 2.0 as real);
+    let rotated = q * Vector3::new(1.0, 4.2, 2.0);
+    assert_eq_approx!(rotated.x, -2.0);
+    assert_eq_approx!(rotated.y, 4.2);
+    assert_eq_approx!(rotated.z, 1.0);
+}
```

Comment:
Could you make one of these 2 tests a non-90° rotation? Maybe run the same in GDScript and check the values. Thanks a lot! :slightly_smiling_face:

Reply:
Added a non-90° and checked back in godot to make sure numbers are correct.